MatrixSDK/Crypto/SecretStorage/MXSecretStorage.m

```diff
@@ -126,6 +127,104 @@ - (MXHTTPOperation*)createKeyWithKeyId:(nullable NSString*)keyId
     return operation;
 }
 
+- (MXHTTPOperation*)createKeyWithKeyId:(nullable NSString*)keyId
+                               keyName:(nullable NSString*)keyName
+                            passphrase:(nullable NSString*)passphrase
+                               success:(void (^)(MXSecretStorageKeyCreationInfo *keyCreationInfo))success
+                               failure:(void (^)(NSError *error))failure
+{
+    MXLogDebug(@"[MXSecretStorage] createKeyWithKeyId: Creating new key with passphrase");
+    keyId = keyId ?: [[NSUUID UUID] UUIDString];
+    
+    MXHTTPOperation *operation = [MXHTTPOperation new];
+    
+    MXWeakify(self);
+    dispatch_async(processingQueue, ^{
+        MXStrongifyAndReturnIfNil(self);
+        
+        NSError *error;
+        
+        NSData *privateKey;
+        MXSecretStoragePassphrase *passphraseInfo;
+        
+        if (passphrase)
+        {
+            // Generate a private key from the passphrase
+            NSString *salt;
+            NSUInteger iterations;
+            privateKey = [MXKeyBackupPassword generatePrivateKeyWithPassword:passphrase
+                                                                        salt:&salt
+                                                                  iterations:&iterations
+                                                                       error:&error];
+            if (!error)
+            {
+                passphraseInfo = [MXSecretStoragePassphrase new];
+                passphraseInfo.algorithm = @"m.pbkdf2";
+                passphraseInfo.salt = salt;
+                passphraseInfo.iterations = iterations;
+            }
+        }
+        else
+        {
+            OLMPkDecryption *decryption = [OLMPkDecryption new];
```

Comment:
This uses OLMKit

Reply:
> This uses OLMKit
`
I think it's just generating 32 random bytes, you can replace with `SecRandomCopyBytes`

Reply:
@BillCarsonFr okay I think I have replaced it correctly, let me know if this is fine and we can merge it